lib/std/fmt.zig

```diff
@@ -560,13 +560,25 @@ fn formatFloatValue(
     options: FormatOptions,
     writer: anytype,
 ) !void {
+    // this buffer should be enough to display all decimal places of a decimal f64 number.
+    var buf: [512]u8 = undefined;
+    var buf_stream = std.io.fixedBufferStream(&buf);
+
     if (fmt.len == 0 or comptime std.mem.eql(u8, fmt, "e")) {
-        return formatFloatScientific(value, options, writer);
+        formatFloatScientific(value, options, buf_stream.writer()) catch |err| switch (err) {
+            error.NoSpaceLeft => unreachable,
+            else => |e| return e,
+        };
     } else if (comptime std.mem.eql(u8, fmt, "d")) {
-        return formatFloatDecimal(value, options, writer);
+        formatFloatDecimal(value, options, buf_stream.writer()) catch |err| switch (err) {
+            error.NoSpaceLeft => unreachable,
+            else => |e| return e,
+        };
     } else {
         @compileError("Unknown format string: '" ++ fmt ++ "'");
     }
+
+    return formatBuf(buf[0..buf_stream.pos], options, writer);
```

Comment:
```suggestion
    return formatBuf(buf_stream.getWritten(), options, writer);
```

Reply:
I searched for that function but i didn't find it :facepalm: 
Good idea!